res/shader/surface.inc.wgsl

```diff
@@ -40,6 +51,26 @@ fn modulo(a: i32, b: i32) -> i32 {
     return select(c, c+b, c < 0);
 }
 
+fn get_storage_index(ipos: vec2<i32>) -> StorageIndex {
+    let x = modulo(ipos.x, i32(u_Surface.texture_scale.x));
+    let y = modulo(ipos.y, i32(u_Surface.texture_scale.y));
+    // let x = ipos.x;
+    // let y = ipos.y;
+    let index = u32(y * i32(u_Surface.texture_scale.x) + x);
+
+    var si: StorageIndex;
+    si.index = index / 4u;
+    si.shift = 8u * (index % 4u);
+    si.mask = 0xFFu << si.shift;
```

Comment:
You don't actually need both shift and mask. Instead of doing `(something & mask) >> shift` you can just do `(something >> shift) & 0xFFu` all the time